kernel/src/main/java/org/kframework/compile/ResolveContexts.java

```diff
@@ -176,7 +176,13 @@ public void apply(KApply k) {
         List<ProductionItem> items = new ArrayList<>();
         KLabel freezerLabel;
         if (cooled instanceof KApply) {
-            freezerLabel = getUniqueFreezerLabel(input, ((KApply)cooled).klabel().name() + finalHolePosition[0]);
+            KApply kApply = (KApply)cooled;
+            String name = kApply.klabel().name();
+            K firstArg = kApply.klist().items().get(0);
```

Comment:
Can we move this declaration inside the name.equals condition? I'm a little worried about the possibility of an IndexOutOfBounds exception.